src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1136,6 +1139,16 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
             }
         }
 
+        private bool CanApplyParseOptionChange(ParseOptions old, ParseOptions @new)
+        {
+            // only changes to C# LanguageVersion are supported at this point
```

Comment:
Hum. The trouble I have with moving this is that in `VisualStudioWorkplaceImpl` I no longer am able to implement this check, because only the new `ParseOptions` are available...
I'll need to think more about this.